### ISSUE TITLE
Verify and improve tutorial

### ISSUE THREAD

**Issue by szhorvat** [igraph/igraph#1676]
The igraph tutorial ([here online](https://igraph.org/c/doc/igraph-Tutorial.html), [source file here](https://github.com/igraph/igraph/blob/master/doc/tutorial.xml)) should be reviewed:

 - Is it up to date?
 - Is it understandable by a newcomer?
 - Are essential points unexplained or missing?

Ideally, this should be done by someone who is new to using igraph, to get a fresh perspective. Thus, marking this as `good-first-issue`. **Contributions are very welcome!**

**Comment by lucyharlow** [comment]
hello! This sounds like a great first issue, may I take a look? I'm new to igraph -- in fact new to a lot of things right now so I'm VERY used to reading tutorials :)

**Comment by szhorvat** [comment]
Yes, please go ahead. Please let us know what operating system you are using.

The first thing you need to do is learn how to compile and install igraph: https://igraph.org/c/doc/igraph-Installation.html  Feedback on this is welcome as well.

At some point, you may want to build your own version of the documentation (instead of using the online one) in order to: (1) be able to read the latest development version (2) be able to edit it.  Let's leave this as a later step, as you will need to install a few special tools, and you would _probably_ need Linux or macOS (I'm not sure if documentation building will work on Windows).

If you need help, feel free to post on our forum at https://igraph.discourse.group/ and ping me as `@szhorvat`

Please note that this week is unusually busy for all of us, so we may be very slow to respond until maybe mid-next-week.

**Comment by lucyharlow** [comment]
Hi! I will get to it tomorrow, looking forward to it! I'm using macOS Big Sur version 11.2, though I have access to a Linux computer as well.

**Comment by szhorvat** [comment]
macOS is perfectly fine. The easiest way to install tools you need is MacPorts (I used this) or Homebrew (some other igraph developers use this).

**Comment by HarshKumarSingh07** [comment]
HI, I am interested in this particular issue. Please let me know how to begin. As I am new to igraph so I am currently going through igraph tutorials docs (https://igraph.org/c/doc/igraph-Tutorial.html). And I am using **Windows** as my operating system.

**Comment by szhorvat** [comment]
@HarshKumarSingh07 You are welcome to work on this. What I wrote above still applies, but in the meantime we realized that the compilation instructions are a bit outdated. To make things simpler, I recommend compiling igraph as a shared library first.

Please post on the forum (https://igraph.discourse.group/) if you get stuck, and describe the problem in as much detail as possible. This feedback will be useful for improving the tutorial.

On Windows, you can either use Visual Studio or MSYS2/MinGW. Do note that none of us currently work on Windows, I am not sure if anyone would be able to help with issues that arise while using Visual Studio's graphical interface (personally, I use only the command line tools when I test on Windows).

**Comment by HarshKumarSingh07** [comment]
Hey @szhorvat sir, Sorry to say I tried but I am getting problem in setting up the igraph library on my windows system. Can you guide or share some relevant work which could help me out in compiling the igraph library on my machine?

**Comment by szhorvat** [comment]
Have you already looked at https://igraph.org/c/doc/igraph-Installation.html ?  If yes, describe in detail how far you got, and where you got stuck, but please use the forum (https://igraph.discourse.group/), not the issue tracker.

**Comment by szhorvat** [comment]
For those interested in this issue, right now easiest way to compile your first program is to use the `master` branch from the igraph git repository, and use CMake for your own program as well.

Assuming your first program is in the file `igraph_test.c`, your `CMakeLists.txt` file may look as follows:

```
cmake_minimum_required(VERSION 3.16)

project(igraph_find_test C)

find_package(igraph REQUIRED)

add_executable(igraph_test igraph_test.c)
target_link_libraries(test PUBLIC igraph::igraph)
```

If igraph was installed at a non-standard prefix (`CMAKE_INSTALL_PREFIX`, see [installation docs](https://igraph.org/c/doc/igraph-Installation.html)) then use `-DCMAKE_PREFIX_PATH=...` with the same prefix when compiling your own program.

**Comment by szhorvat** [comment]
Draft update to the tutorial: https://igraph.discourse.group/t/igraph-tutorial-update/708

**Comment by szhorvat** [comment]
@HarshKumarSingh07 @lucyharlow Are you still interested in this? Version 0.9.2 was released today, with improvements that make it easier to get started. See the updated installation instructions and tutorial in the documentation.

**Comment by HarshKumarSingh07** [comment]
Yes @szhorvat , I am going through it right now. Will back to you if any problem would be there.

**Comment by OlofMorra** [comment]
Hi all, 

I want to learn C and contribute to an open-source project. As Data Science student igraph looks like an interesting repository to contribute to! I use a MacBook Pro with Big Sur 11.2.3. I will start the tutorial and comment on the understandability! 

Although I am a newcomer to C (used it once), I am fairly experienced with programming practices and some other languages, so I do have considerate knowledge on the algorithms in use and programming itself.  

I am looking forward to learning new stuff!

**Comment by isakhagg** [comment]
Hi, I'm a master's student in computer sience and engineering at Linköping University. Currently I'm looking into getting started with open source development. Is iGraph a good first project to start in? I have knowledge in both C, C++ and Python.

**Comment by szhorvat** [comment]
@isakhagg igraph welcomes new contributors. For beginners, this issue is the best way to get started. In other words, go through the tutorial, write your first working program, and give feedback about what needs to be improved in the tutorial/documentation.  You can ask for help in the forum, https://igraph.discourse.group/  When you are done with this, we can move on to #1585, then gradually to more complex and more interesting tasks.

**Comment by szhorvat** [comment]
Adding to 0.10 milestone. The aim is to make sure that the tutorial is up to date with 0.10, does not use deprecated functionality and all sample code compiles.

**Comment by ntamas** [comment]
AFAIK we already compile and test `examples/` as part of the regular unit test suite so any deprecations should have been already caught there. We only need to go through code that is inlined in the documentation (if any).

**Comment by szhorvat** [comment]
@ntamas I'll give it a quick read to see if there's anything that needs to be fixed.

**Comment by JRickenbacher** [comment]
Hey, I've read through the above comments. If this issue is still open, I would like to help if possible.

I might need to brush up on my C, but I'd like to get started helping with open source projects and I'm very interested in graphs in general, so I thought this might be a good place to start.

I mainly use MacOS currently, but I also have a Linux machine if you would prefer I test on that.

I'll go through the installation steps (https://igraph.org/c/doc/igraph-Installation.html) and tutorial (https://igraph.org/c/doc/igraph-Tutorial.html). Are those the most current links?

Is there anything in particular you would like feedback on, or just how readable everything is from a first time user?

**Comment by szhorvat** [comment]
Yes, feedback is welcome.

Please use the `develop` branch in the GitHub repository. This will soon become version 0.10. The documentation is here:

https://igraph.org/c/html/develop/

The documentation version can be selected in the page header on the left. It should be "develop" and not "latest".

**Comment by Jeroenvh99** [comment]
in the paragraph about compiling with CMake it says the following:
"C++ support is enabled by default when no languages are explicitly specified in CMake's project command. If you do specify some languages explicitly, make sure to also include CXX."
this leaves me wondering where to include CXX, I think it should be added to the list of languages, but I think it would be more clear to say "make sure to also include CXX to the languages".

**Comment by szhorvat** [comment]
I agree that this isn't really tutorial material. 

If you use `project(something)`, as recommended in the current version of the tutorial, all is fine, as both C and C++ are enabled by default. If you explicitly enable C with `project(something C)` then C++ will not be enabled. `project(something C CXX)` is fine. Old versions of the tutorial had `project(something C)`.
